src/modules/asset-segments/asset-segments.service.ts

```diff
@@ -0,0 +1,67 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { AssetSegmentsEntity, AssetSegmentsDto } from 'lib-typeorm';
+
+
+@Injectable()
+export class AssetSegmentsService {
+  constructor(
+    @InjectRepository(AssetSegmentsEntity)
+    private readonly assetSegmentsRepository: Repository<AssetSegmentsEntity>,
+  ) {}
+
+  async create(assetSegmentsDto: AssetSegmentsDto): Promise<AssetSegmentsEntity> {
+    const assetSegment = this.assetSegmentsRepository.create(assetSegmentsDto);
+    return this.assetSegmentsRepository.save(assetSegment);
+  }
+
+  async findAll(query: any): Promise<AssetSegmentsEntity[]> {
+    const queryBuilder = this.assetSegmentsRepository.createQueryBuilder('asset_segments');
+    
+    if (query.isin) {
+      queryBuilder.andWhere('asset_segments.isin = :isin', { isin: query.isin });
+    }
+
+    if (query.label) {
+      queryBuilder.andWhere('asset_segments.label LIKE :label', { label: `%${query.label}%` });
+    }
+
+    if (query.value) {
+      queryBuilder.andWhere('asset_segments.value = :value', { value: query.value });
+    }
+
+    if (query.metric) {
+      queryBuilder.andWhere('asset_segments.metric = :metric', { metric: query.metric });
+    }
+
+    if (query.segmentType) {
+      queryBuilder.andWhere('asset_segments.segmentType = :segmentType', { segmentType: query.segmentType });
+    }
+
+    if (query.periodicity) {
+      queryBuilder.andWhere('asset_segments.periodicity = :periodicity', { periodicity: query.periodicity });
+    }
```

Comment:
why we are using querybuilder here with so many if statements? Is it possible to do this with TypeORM query? Let me know if I am missing something.

Reply:
Used an alternative method now. Now i am using an array that stores the conditions to build dynamic queries